src/pg_ext/big_decimal.cr

```diff
@@ -0,0 +1,21 @@
+require "big"
+
+module PG
+  struct Numeric
+    # Returns a BigDecimal representation of the numeric. This retains all precision.
+    def to_big_d
+      return BigDecimal.new(0) if nan? || ndigits == 0
+      BigDecimal.new(to_s)
```

Comment:
There is `pg/pg_ext/big_rational` so this method can become:

```crystal
def to_big_d : BigDecimal
  to_big_r.to_big_d
end
```

In fact, maybe both files could be merged into a single `pg/pg_ext/big` file?